internal/docstore/firedocstore/query.go

```diff
@@ -349,3 +356,44 @@ func fieldRef(fp []string) *pb.StructuredQuery_FieldReference {
 func (c *collection) QueryPlan(q *driver.Query) (string, error) {
 	return "unknown", nil
 }
+
+// For delete and update queries, limit the number of write actions per RPC, to bound
+// client memory.
+// This is a variable so it can be modified for tests.
+var maxWritesPerRPC = 1000
+
+func (c *collection) RunDeleteQuery(ctx context.Context, q *driver.Query) error {
+	q.FieldPaths = [][]string{{"__name__"}}
+	iter, err := c.newDocIterator(ctx, q)
+	if err != nil {
+		return err
+	}
+	defer iter.Stop()
+
+	opts := &driver.RunActionsOptions{}
+	var pws []*pb.Write
+	for {
+		res, err := iter.nextResponse(ctx)
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			return err
+		}
+		pws = append(pws, &pb.Write{
+			Operation: &pb.Write_Delete{Delete: res.Document.Name},
+		})
+		if len(pws) >= maxWritesPerRPC {
```

Comment:
Is there a limit on the client library? Why do we want to limit the number of writes here? More efficient?

Reply:
To limit the memory on the client. We don't want to read a million items and then delete them, because the client would have to store all million.

Reply:
I just got a `rpc error: code = InvalidArgument desc = maximum 500 writes allowed per request` when running some benchmarks, I guess there is a limit on the writes.